compiler/rustc_ast/src/ast.rs

```diff
@@ -1339,11 +1339,11 @@ pub enum ExprKind {
     /// and the second field is the list of arguments.
     /// This also represents calling the constructor of
     /// tuple-like ADTs such as tuple structs and enum variants.
-    Call(P<Expr>, Vec<P<Expr>>),
+    Call(P<Expr>, ThinVec<P<Expr>>),
     /// A method call (e.g. `x.foo::<Bar, Baz>(a, b, c)`).
     MethodCall(Box<MethodCall>),
     /// A tuple (e.g., `(a, b, c, d)`).
-    Tup(Vec<P<Expr>>),
+    Tup(ThinVec<P<Expr>>),
```

Comment:
Possibly stupid questions ahead, I don't know much about the AST.
Unit types count as tuple I assume. Would having a separate `Unit()` variant make a difference here? Then the other ones would always be non-empty.

Reply:
There would be no space/speed benefit to having a separate `Unit`. The `Tup` case would still be the same size. In fact, an empty `ThinVec` is the special case -- it's just a single pointer to a static thing -- so there's no point trying to handle it separately.